store/listenkv/store.go

```diff
@@ -35,14 +35,14 @@ func (s *Store) Get(key []byte) []byte {
 func (s *Store) Set(key []byte, value []byte) {
 	types.AssertValidKey(key)
 	s.parent.Set(key, value)
-	s.onWrite(false, key, value)
+	s.listener.OnWrite(s.parentStoreKey, key, value, false)
```

Comment:
possible nil reference here right? since `listener` is a pointer type.

Reply:
> > > > What is in the 22MB file `store/streaming/abci/examples/file/file`, should this be committed?
> > > 
> > > 
> > > @egaxhaj would be good to get more information on this as well.
> > 
> > 
> > These are the go example plugin binaries (stdout, file). I can remove the `file` binary but the `stdout` binary is required to run the unit tests during CI as it is the default.
> 
> lets remove the file and the binary may be harder.

I've removed the file plugin binary and added a readme and .gitignore in that folder. I've resolved the conflicts as well.

Reply:
It is possible only if misconfigured. listenkv stores are attached by checking if `ListeningEnabled` is `true`. `ListeningEnabled` checks to to see if a MemoryListener is `non-nil` for each KeyStore that a listener will be attached. this is a hot code path so let's leave this as is to avoid an extract `if` check for an unlikely case due to misconfiguration.

https://github.com/cosmos/cosmos-sdk/blob/be35b4fdceeaf889471d89f5a4667482ef1aedf1/store/rootmulti/store.go#L416-L421

https://github.com/cosmos/cosmos-sdk/blob/be35b4fdceeaf889471d89f5a4667482ef1aedf1/store/rootmulti/store.go#L520-L522